miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/UploadController.java

```diff
@@ -326,6 +328,15 @@ public void uploadLibraryList(MultipartHttpServletRequest request) throws IOExce
       log.debug("UPLOAD FAIL:", e);
     }
   }
+  
+  @RequestMapping(value = "/servicerecord", method = RequestMethod.POST)
+  public void uploadServiceRecordDocument(MultipartHttpServletRequest request) throws IOException {
+    String recordId = request.getParameter("serviceRecordId");
```

Comment:
Adding check that throws IOException in case of null/missing ID parameter, or non-existant object (Service Record in this case) in this method and other similar methods in UploadController